pkg/kubectl/apis/manifest/v1alpha1/types.go

```diff
@@ -0,0 +1,118 @@
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// Package contains the
+type Package struct {
+	metav1.TypeMeta `json:",inline"`
+	// Standard object's metadata.
+	// More info: https://git.k8s.io/community/contributors/devel/api-conventions.md#metadata
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	// Labels to add to all objects and selectors.
+	// These labels would also be used to form the selector for apply --prune
+	// Named differently than “labels” to avoid confusion with metadata for this object
+	ObjectLabels map[string]string `json:"objectLabels,omitempty"`
+
+	// Annotations to add to all objects.
+	ObjectAnnotations map[string]string `json:"objectAnnotations,omitempty"`
+
+	// Partial name that will prefix the name of the base resources.
+	NamePrefix string `json:"namePrefix,omitempty"`
+
+	// Description of the application package.
+	Description string `json:"description,omitempty"`
+
+	// Search keywords for this application package.
+	Keywords []string `json:"keywords,omitempty"`
+
+	// Version of the application package.
+	AppVersion string `json:"appVersion,omitempty"`
+
+	// Homepage of the application package.
+	Home string `json:"home,omitempty"`
+
+	// All sources where you can find this application package.
+	Sources []string `json:"sources,omitempty"`
```

Comment:
Suggest making this singular
```
Source string
```
so user not forced to code for an array and wonder about ordering / which entry is best.

If strongly desired, add secondary field
```
SourceMirrors []string
```
to allow specification of (alleged) mirrors of Source.

Also, suggest the following field commentary:
```
// Source specifies the URL, e.g. https://github.com/foo/bar.git, file://host/path, etc.
// hosting the resource files specified in Base and Overlays.
Source string

// A Base entry specifies the file path, relative to Source, of an API resource definition.
Bases []string

// An Overlay entry specifies the file path, relative to Source, of a file
// containing a Strategic Merge Patch overlay in API resource form.
Overlay []string
```